### PR TITLE
Aman 141/forgot pw

### PR DESCRIPTION
**Pull request by Scarletra**
# Description  
[describe the problem or user story being addressed]

# Ticket Issue 
[Provide link to Jira](https://a-man-ppl.atlassian.net/browse/AMAN-141?atlOrigin=eyJpIjoiNWJkODk5Yjc0NzFmNGJiNGJiMzkyZTU3NTNmODY2M2QiLCJwIjoiaiJ9)

# Changes Made  
Add new module ForgotPasswordModule and api service to reset password

# Issue Type
- [ ] 🐛 Bug fix 
- [ ] ⚙️ Refactor
- [X] 💻 New feature 
- [ ] 📄 Documentation update
- [ ] ⚠️ Breaking changes
- [X] 🔍 Testing

# Added/updated tests?

- [X] Yes
- [ ] No, and this is why: [please replace this line with details on why tests have not been included]

# Additional Notes  
You can access the page by clicking the lupa kata sandi button at login page or access it directly on /forgot-password

# (optional) What photo/gif best describes this PR or how it makes you feel?

## Summary by CodeRabbit

- **New Features**
  - Introduced a "Forgot Password" feature, allowing users to request a password reset via email.
  - Added a dedicated "Forgot Password" page accessible from the login screen.
  - Implemented an "AuthBanner" component for improved user experience on authentication pages.
  - Updated login page with a direct link to the password recovery page.

- **Tests**
  - Added comprehensive tests for the "Forgot Password" flow, covering UI, validation, API integration, and edge cases.

### CODE REVIEW

**Comment by raviehasan** [comment]
nicee overall udah oke doo sonarnya jg udah aman semua, tapi ini aja sih paling, ini tu harusnya register apa login ya? tulisan di buttonnya `Masuk` tapi hrefnya ke `/register` soalnya
image

File: src/modules/LoginFormModule/Section/login.tsx [hunk 1]
```diff
@@ -64,7 +64,13 @@ export const LoginForm = ({
             />
           </div>
 
-          <div className="mt-6">
+          <Typography variant="p4" className="mt-2 text-rencanakan-sea-blue-500">
+            <a href="/forgot-password" className="hover:text-blue-700 text-blue-500">
+              Lupa kata sandi?
+            </a>
+          </Typography>
```
Comment (Abbilville) [inline src/modules/LoginFormModule/Section/login.tsx:71]: Udah bagus sih, tapi ini dari segi UI nya berarti cuma tulisan gitu kah yang langsung bakal diarahin ke forgot password

Comment (Scarletra) [reply src/modules/LoginFormModule/Section/login.tsx:71]: harusnya iyasih yang paling make sense, klo tambahin di regist agak aneh aja harusnya user tau klo emng dia blm punya akun
